### PR TITLE
Register metadata on classes

### PR DESCRIPTION
**Pull request by revmischa**
This uses [reflect-metadata](https://github.com/rbuckton/reflect-metadata) to stash metadata on classes and methods for generating CDK resources. Seems to work pretty well.

See [the test](https://github.com/jetbridge/jkv2-playground/pull/1/files#diff-95621e17047d633a983ee36284cfba6f833a820bee9e0038c976043d7e75c091) to get an idea of how it should work

The idea is this should generate two lambda functions and three APIGW routes:
1. AlbumCrudApi handler
2. `/album/test` custom route
3. `/blargle` function-based route (no class) - can't use a decorator on a function (I think) but we can simply wrap it in a function

```typescript
@Entity()
export class Album extends BaseModel {
  @Column({ nullable: true })
  title: string;
}

@CrudApi({ model: Album, route: "/album", memorySize: 512 })
export class AlbumCrudApi extends CrudApiBase {
  @SubRoute("/test")
  async test() {
    return "Testerino";
  }

  post: APIGatewayProxyHandlerV2 = async () => "Posterino";
}

Route({ route: "/blargle" })(async function (event) {
  return JSON.stringify({
    message: "function route",
    rawQueryString: event.rawQueryString,
  });
});
```

Tests pass but CI is broken because of [this dumb CDK issue](https://github.com/aws/aws-cdk/issues/12354)

### CODE REVIEW

File: packages/backend/src/api/topic.ts [hunk 1]
```diff
@@ -14,10 +13,22 @@ export class Topic extends BaseModel {
   name: string;
 }
 
-@RegisterCrudApi(app, { model: Topic, route: "/topic", memorySize: 512 })
+@CrudApi({ model: Topic, route: "/topic", memorySize: 512 })
```
Comment (Dizzzmas) [inline packages/backend/src/api/topic.ts:16]: How do we stick schema validation on top of it?

Use that library to put decorators on top of fields in `Topic`?
https://github.com/typestack/class-validator

File: packages/backend/src/index.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export { app } from "./app";
 import { TopicCrudApi } from "./api/topic";
 
+// a list of resources we would like infrastructure generated for
 export const stackResources = [TopicCrudApi];
```
Comment (Dizzzmas) [inline packages/backend/src/index.ts:5]: We need to import `TopicCrudApi`, but don't need to do so with the function wrapped in `Route({ route: "/blargle" })`?

**Review by Dizzzmas — APPROVED** [review]
Looks lit.

Wonder what other potential uses of reflect-metadata there are.

File: packages/backend/src/api/topic.ts (hunk 1 above)
Comment (revmischa) [reply packages/backend/src/api/topic.ts:16]: Yeah that's one option and would fit nicely. Would like to look into io-ts again

File: packages/backend/src/index.ts (hunk 2 above)
Comment (revmischa) [reply packages/backend/src/index.ts:5]: No you're right we do, just haven't gotten that far yet

File: packages/backend/src/api/topic.ts (hunk 1 above)
Comment (Dizzzmas) [reply packages/backend/src/api/topic.ts:16]: There's a library mainly for form validation, that allows to create a schema and then validate user input against it.

If we can stick it on top of the API could kill 2 birds with 1 stone.

https://github.com/jquense/yup

**Pull request merged**